src/profiler/model.ts

```diff
@@ -73,35 +93,46 @@ export interface RowToolbarOptions {
 }
 
 export interface ProfilerToolbarOptions {
-  orientOptions?: {
-    type?: Orient
-    isCheckboxShown?: boolean
-    onOrientChange?: (orient: Orient) => void
-  }
-  jsonOptions?: {
-    checked?: boolean
-    isCheckboxShown?: boolean
-    onChange?: (shown: boolean) => void
-  }
+  orientOptions?:
+    | boolean
+    | {
+        type?: Orient
+        isCheckboxShown?: boolean
+        onOrientChange?: (orient: Orient) => void
+      }
+  jsonOptions?:
+    | boolean
+    | {
+        checked?: boolean
+        isCheckboxShown?: boolean
+        onChange?: (shown: boolean) => void
+      }
   // TODO: too much optional stuff?
-  normalizeOptions?: {
-    checked?: boolean
-    isCheckboxShown?: boolean
-  }
-  searchOptions?: {
-    disabled?: boolean
-    onChange?: (value: string) => void
-  }
+  normalizeOptions?:
+    | boolean
+    | {
+        checked?: boolean
+        isCheckboxShown?: boolean
+      }
+  searchOptions?:
+    | boolean
+    | {
+        disabled?: boolean
+        onChange?: (value: string) => void
+      }
 }
 
 export interface DataProfilerProps extends AntdTableProps<any> {
   datasets: Datasets
-  profilerToolbarOptions?: null | false | ProfilerToolbarOptions
-  colors?: Array<string>
-  rowToolbarOptions?: null | false | RowToolbarOptions
+  config?: {
+    profilerToolbarOptions?: null | false | ProfilerToolbarOptions
+    rowToolbarOptions?: null | false | RowToolbarOptions
+    colors?: Array<string>
+    smallSize?: boolean
+    visibleColumns?: Array<string>
+    pagination?: false
+  }
```

Comment:
you may like instead `pagination?: PaginationProps | false` :

```
 enablePagination?: boolean
 pagination?: PaginationProps
```

Reply:
we should be able to pass the same pagination model as in antd - https://ant.design/components/table/#API

Reply:
```
  const _pagination =
    typeof props.config?.pagination === 'boolean' ? props.config.pagination : pagination
```

In the updated code, the same pagination entity is configured in two places simultaneously `DataProfilerProps` and `DataProfilerProps.config`. It would make sense to keep just one